include/TrustWalletCore/TWCurve.h

```diff
@@ -13,6 +13,7 @@ TW_EXTERN_C_BEGIN
 TW_EXPORT_ENUM()
 enum TWCurve {
     TWCurveSECP256k1       /* "secp256k1" */,
+    TWCurveSECP256k1groestl/* "secp256k1-groestl" */,
```

Comment:
Why do we need this? Looks like it's a different hash not a different curve.

Reply:
For trezor-crypto to generate correct xpub. TrustWallet uses trezor-crypto directly for xpub generation and trezor-crypto determines correct hashers using curve name from this struct:
https://github.com/TrustWallet/wallet-core/blob/5ad008827a16036cbdf2475a1872face225d6a62/trezor-crypto/src/secp256k1.c#L74-L81

Reply:
Hmm, seems weird. I'd rather pass both the curve and the hasher separately.

Reply:
By the way, I like your hasher function approach.

Reply:
Yes, I agree that it is a bit weird. But this is how Decred was implemented in trezor-crypto, so I just followed existing architecture to add Groestlcoin. I guess it was done to minimize API changes, adding argument with set of hashers to all functions would require a lot of changes to codebase, because there is no optional parameters in pure-C.

Reply:
Trezor-crypto is C, wallet core is C++. Let's use C++.

Reply:
How do you propose to fix this? TWCurve is used just to get curve name and pass to trezor-crypto? All xpub generation is done inside trezor-crypto.

Reply:
As I said, have two parameters in the C++ function. The curve and the hash function. Use, modify or replace trezor crypto as needed.

Reply:
If you find this change is out of scope for this PR I can help with this.

Reply:
Yes, could you please help with this? Because I do not really understand how you propose to decouple curve from hashers without stopping using trezor-crypto all together, or seriously changing its API which will be maintance hell for future updates from upstream.

Reply:
Ok I looked into it. You are right that it's a bit of work. Requires rewriting bip32.c. But I searched your PR for `TWCurveSECP256k1groestl` and you only use it in a test. Can we leave it out for now?

Reply:
I made some of the changes here: https://github.com/TrustWallet/wallet-core/pull/252

Reply:
Yes, we can use SECP256k1 curve if the wallet will not display xpub to user or use xpub in requests to a blockbook server (but I thought you are going to use new xpub feature from blockbook).

Reply:
Let's do that for now and I'll spend a bit more time figuring this out.

Reply:
@yura-pakhuchiy Trust Wallet does display xpub/zpub for Bitcoin/Bitcoin Cash/Dash/Litecoin/Zcash/Zcoin (Export Account Public Keys). 
Does this mean the zpub shown on Export Account Public Keys will not be valid for Groestlcoin?
Shouldn't the Decred implementation on Trust Wallet have the same issue (invalid xpub/zpub) as us?

Reply:
@gruve-p Oops, I have not noticed that the wallet does display xpub to user (thought it is used only internally). I think it will be bad user experience if the wallet will show incorrect xpubs. @alejandro-isaza maybe we can keep this curve name in meantime (it is just a few lines of code)?

Reply:
Yes, Decred would be wrong. I'll definitely work on fixing this. Let's hold off on merging this PR then until it's fixed.

Reply:
Ok, this should do it: https://github.com/TrustWallet/wallet-core/commit/581e828ccdcf2c52219939871b3b9301f23eb463